src/errors.rs

```diff
@@ -386,6 +386,19 @@ fn valid_profile_names() -> String {
         .join(", ")
 }
 
+fn remove_component_msg(cs: &Component, manifest: &Manifest, toolchain: &str) -> String {
+    format!(
+        "rustup component remove --toolchain {}{} {}",
+        toolchain,
+        if let Some(target) = cs.target.as_ref() {
+            format!(" --target {}", target)
+        } else {
+            String::default()
+        },
+        cs.short_name(manifest)
+    )
+}
+
```

Comment:
Can you please give me some examples of how this would look when the component in question has a target which is the same as the toolchain?  e.g. if nightly-x86_64-unknown-linux-gnu is missing rustfmt-x86_64-unknown-linux-gnu.

Reply:
With the current changes, it would look like this:

`rustup component remove --toolchain nightly-x86_64-unknown-linux-gnu --target rustfmt-x86_64-unknown-linux-gnu`

Reply:
Yeah, I'd expect `rustup component remove --toolchain nightly-x86_64-unknown-linux-gnu rustfmt` I'm not sure that `--target` makes any sense here, thinking about it some more.

In fact, now I think more, I expect there's really two situations we need to deal with:

1. A "normal" component needs removing (no target is needed)
2. `rust-std` needs removing for a target which doesn't match the toolchain host, in which case we ought to recommend `rustup target remove --toolchain $TOOLCHAIN $TARGET` e.g. `rustup target remove --toolchain nightly-x86_64-unknown-linux-gnu wasm32-unknown-unknown`

Reply:
In addition, as mentioned in #2440 we should check if we need to use the rename tables here.

Reply:
What are the rename tables?

Reply:
A toolchain channel has a small mapping for renaming components.  e.g. the underlying component for rustfmt is actually rustfmt-preview and we have a rename in place to rename that to rustfmt for users.  If you look at the end of a channel manifest you'll see:

```toml
[renames.clippy]
to = "clippy-preview"

[renames.miri]
to = "miri-preview"

[renames.rls]
to = "rls-preview"

[renames.rustfmt]
to = "rustfmt-preview"
```

We *might* need to look at using https://github.com/rust-lang/rustup/blob/master/src/dist/manifest.rs#L313 I'm not sure.

I guess that to be sure on that part we ought to ensure we're using `rls` in one of the tests since IIRC that is renamed in the test manifests.

Reply:
I've been through the codepath and believe it is using the rename tables so that's fine.

I've also written a patch which special-cases `rust-std` to suggest using `rustup target remove`.